sgkit/tests/test_ibs.py

```diff
@@ -0,0 +1,153 @@
+        ds["call_allele_count"] = (
+            ds.call_allele_count.dims,
+            ds.call_allele_count.data.rechunk(chunks),
+        )
+    ds = pairwise_ibs(ds)
+    actual = ds.ibs_probabilities.values
+    expect = np.array(
+        [
+            [[1.0, 0.0, 0.5], [0.0, 1.0, 0.5], [0.5, 0.5, 0.5]],
+            [[1.0, 1.0, 0.5], [1.0, 1.0, 0.5], [0.5, 0.5, 0.5]],
+        ]
+    )
+    np.testing.assert_array_equal(expect, actual)
+
+
+@pytest.mark.parametrize(
+    "chunks",
+    [
+        None,
+        ((2,), (3,), (3,)),
+        ((1, 1), (1, 2), (3,)),
+        ((1, 1), (1, 2), (2, 1)),
+        ((1, 1), (1, 1, 1), (1, 1, 1)),
+    ],
+)
+def test_pairwise_ibs__tetraploid_multiallelic(chunks):
+    ds = simulate_genotype_call_dataset(
+        n_variant=2,
+        n_sample=3,
+        n_ploidy=4,
+        n_allele=3,
+        seed=0,
+    )
+    ds = count_call_alleles(ds)
+    ds.call_genotype.data[0, 2] = -1  # null call
+    if chunks is not None:
+        ds["call_allele_count"] = (
+            ds.call_allele_count.dims,
+            ds.call_allele_count.data.rechunk(chunks),
+        )
+    ds = pairwise_ibs(ds)
+    actual = ds.ibs_probabilities.values
+    expect = np.array(
+        [
+            [[0.5, 0.375, np.nan], [0.375, 0.375, np.nan], [np.nan, np.nan, np.nan]],
+            [[1.0, 0.25, 0.0], [0.25, 0.625, 0.1875], [0.0, 0.1875, 0.625]],
+        ]
+    )
+    np.testing.assert_array_equal(expect, actual)
+
+
+@pytest.mark.parametrize(
+    "chunks",
+    [
+        ((7,), (5,), (2,)),
+        ((2, 7, 3), (7, 5, 6, 2), (3,)),
+        ((20, 20, 11), (7, 7, 7, 7, 5), (7, 7, 7)),
+    ],
+)
+@pytest.mark.parametrize("ploidy", [2, 4])
+@pytest.mark.parametrize("seed", [0, 1])
+def test_pairwise_ibs__reference_implementation(ploidy, chunks, seed):
+    ds = simulate_genotype_call_dataset(
+        n_variant=sum(chunks[0]),
+        n_sample=sum(chunks[1]),
+        n_ploidy=ploidy,
+        n_allele=sum(chunks[2]),
+        missing_pct=0.2,
+        seed=seed,
+    )
+    ds = count_call_alleles(ds)
+    ds["call_allele_count"] = (
+        ds.call_allele_count.dims,
+        ds.call_allele_count.data.rechunk(chunks),
+    )
+    ds = pairwise_ibs(ds)
+    actual = ds.ibs_probabilities.values
+    # reference implementation
+    AF = ds.call_allele_frequency.data
+    expect = (AF[..., None, :, :] * AF[..., :, None, :]).sum(axis=-1).compute()
+    np.testing.assert_array_equal(expect, actual)
+
+
+@pytest.mark.parametrize(
+    "sim,chunks",
+    [
+        [1, ((1000,), (16,), (2,))],
+        [1, ((100,) * 10, (16,), (2,))],
+        [1, ((100,) * 10, (4, 4, 4, 4), (2,))],
+        [1, ((270, 330, 400), (7, 5, 4), (1, 1))],
+        [2, ((1000,), (3,), (2,))],
+        [2, ((100,) * 10, (3,), (2,))],
+        [2, ((100,) * 10, (2, 1), (1, 1))],
+    ],
+)
+def test_Weir_Goudet_beta__hierfstat(sim, chunks):
+    # load reference data calculated with function beta.dosage
+    # from hierfstat R library
```

Comment:
It would be good to add a note somewhere (here or next to the data file) saying how the reference data was generated.